velox/core/CancelPool.h

```diff
@@ -228,9 +229,13 @@ class CancelPool {
   }
 
   std::mutex mutex_;
```

Comment:
@oerling If we have majority of members atomic now, then what does we need "mutex_" for?

Also, it is rather dangerous exposing internal mutex via "std::mutex* mutex()" member function.
I'm afraid we might have some deadlock or race problems in the future.

Should we try to carefully make this class (use) more clear in terms of locks?